lib/packager.js

```diff
@@ -365,15 +366,30 @@ function makeNodejsPackage(opts, done) {
     tasks.push(expand(tmpl, dst, opts.packageInfo));
   });
 
-  async.series(tasks, function(err) {
-    if (!err && opts.copyables.indexOf('PUBLISHING.md') != -1) {
-      console.log('The nodejs package', pkgName(opts.packageInfo),
-                  'was created in', opts.top);
-      console.log('To publish it, read', path.join(opts.top, 'PUBLISHING.md'),
-                  'for the next steps');
+  function runTask(err, jsFiles) {
+    if (err) {
+      done(err);
+      return;
     }
-    done(err);
-  });
+    if (opts.mockApiFilesForTest) {
+      jsFiles = opts.mockApiFilesForTest;
+    }
+    opts.packageInfo.apiFiles = _.map(jsFiles, function(jsFile) {
+      return jsFile.replace(/\.js$/, '');
+    });
+    async.series(tasks, function(err) {
+      if (!err && opts.copyables.indexOf('PUBLISHING.md') != -1) {
+        console.log('The nodejs package', pkgName(opts.packageInfo),
+                    'was created in', opts.top);
+        console.log('To publish it, read', path.join(opts.top, 'PUBLISHING.md'),
+                    'for the next steps');
+      }
+      done(err);
+    });
+  }
+
+  // find lib/*.js files for gax API packages to generate index.js properly.
+  glob.glob('*.js', {cwd: path.join(opts.top, 'lib')}, runTask);
```

Comment:
It looks like `glob` will access the file system even if `opts.mockApiFilesForTest` is specified. Can we skip this step if it is?